packages/platform-core/src/telemetry/account_id_fetcher.ts

```diff
@@ -0,0 +1,22 @@
+import { GetCallerIdentityCommand, STSClient } from '@aws-sdk/client-sts';
+
+/**
+ * Retrieves the account ID of the user
+ */
+export class AccountIdFetcher {
+  /**
+   * constructor for AccountIdFetcher
+   */
+  constructor(private readonly stsClient = new STSClient()) {}
+  fetch = async () => {
+    const stsResponse = await this.stsClient.send(
+      new GetCallerIdentityCommand({})
+    );
+    if (stsResponse && stsResponse.Account) {
+      return stsResponse.Account;
+    }
```

Comment:
If this type is to remain public then I'd rather just use stsClient directly in usage data. It doesn't seem to absorb a lot of complexity.
If this becomes internal it's not a problem.

Reply:
this is not exported outside of platform-core. Is it considered public? Right now it helps with making testing easier.

Reply:
Added the factory to hide this